test/ts/unit/collateralized_contract.ts

```diff
@@ -106,33 +130,29 @@ contract("CollateralizedContract (Unit Tests)", async (ACCOUNTS) => {
             )).to.eventually.be.rejectedWith(REVERT_ERROR);
         });
 
-        it("should throw if the agreement already has already been collateralized", async () => {
-            // We first colletaralize the contract.
-            await collateralContract.collateralize.sendTransactionAsync(
-                ARBITRARY_AGREEMENT_ID,
-                mockToken.address,
-                new BigNumber(5),
-                new BigNumber(moment().add(2, 'years').unix()),
-                { from: COLLATERALIZER }
-            );
-
-            // a second attempt to collateralize the contract should fail.
+        it("should throw if the collateralizer does not have sufficient balance", async () => {
             await expect(collateralContract.collateralize.sendTransactionAsync(
                 ARBITRARY_AGREEMENT_ID,
                 mockToken.address,
-                new BigNumber(10),
-                new BigNumber(moment().add(3, 'years').unix()),
+                COLLATERAL_AMOUNT,
+                new BigNumber(moment().add(2, 'years').unix()),
                 { from: COLLATERALIZER }
             )).to.eventually.be.rejectedWith(REVERT_ERROR);
         });
 
-        it("should throw if the collateral fails to transfer", async () => {
+        it("should throw if the custodian is not allotted sufficient allowance by the collateralizer", async () => {
+            // set a sufficient balance so that we fall through to the allowance check.
```

Comment:
The allowance / balance tests make assumptions about the order in which those two invariants are checked in the terms contract -- for instance, these tests would silently pass for the wrong reasons if you called `mockAllowanceFor` below.

Each test should individually mock a sufficient allowance / insufficient balance, and vice versa.